abi-versions/v0.2.1/README.md

```diff
@@ -0,0 +1,1996 @@
+  - `i32 (uint32_t) status_code`
+* returns:
+  - none
+
+Called when gRPC call or stream `call_id` opened using
+[`proxy_grpc_call`] or [`proxy_grpc_stream`] is received.
+
+gRPC status message can be retrieved using [`proxy_get_status`].
+
+
+## Shared Key-Value Store
+
+### Functions exposed by the host
+
+#### `proxy_set_shared_data`
+
+* params:
+  - `i32 (const char *) key_data`
+  - `i32 (size_t) key_size`
+  - `i32 (const uint8_t *) value_data`
+  - `i32 (size_t) value_size`
+  - `i32 (uint32_t) cas`
+* returns:
+  - `i32 (`[`proxy_status_t`]`) status`
+
+Sets shared data identified by the key (`key_data`, `key_value`)
+to the value (`value_data`, `value_size`).
+
+If the compare-and-swap value (`cas`) is set to a non-zero value,
+then it must match the host's compare-and-swap value in order for
+the update to succeed.
+
+Returned `status` value is:
+- `OK` on success.
+- `CAS_MISMATCH` when `cas` doesn't match host's compare-and-swap
+  value.
+- `INVALID_MEMORY_ACCESS` when `key_data`, `key_size`, `value_data`,
+  `value_size` and/or `cas` point to invalid memory address.
+
+
+#### `proxy_get_shared_data`
+
+* params:
+  - `i32 (const char *) key_data`
+  - `i32 (size_t) key_size`
+  - `i32 (uint8_t **) return_value_data`
+  - `i32 (size_t *) return_value_size`
+  - `i32 (uint32_t *) return_cas`
+* returns:
+  - `i32 (`[`proxy_status_t`]`) status`
+
+Returns shared value (`return_value`) identified by the key (`key_data`,
+`key_value`).
+
+The compare-and-swap value (`return_cas`) can be used for atomically
+updating this value using [`proxy_set_shared_data`].
+
+Returned `status` value is:
+- `OK` on success.
+- `NOT_FOUND` when the requested key was not found.
+- `INVALID_MEMORY_ACCESS` when `key_data`, `key_size`,
+  `return_value_data`, `return_value_size` and/or `return_cas`
+  point to invalid memory address.
+
+
+## Shared Queues
+
+### Functions exposed by the host
+
+#### `proxy_register_shared_queue`
+
+* params:
+  - `i32 (const char *) name_data`
+  - `i32 (size_t) name_size`
+  - `i32 (uint32_t *) return_queue_id`
+* returns:
+  - `i32 (`[`proxy_status_t`]`) status`
+
+Registers shared queue under a name (`name_data`, `name_size`).
+
+If the named queue already exists, then it's going to be opened
+instead of creating a new empty queue.
+
+Items can be enqueued/dequeued on the created/opened queue using
+[`proxy_enqueue_shared_queue`] and/or [`proxy_dequeue_shared_queue`]
+with `return_queue_id`.
+
+Returned `status` value is:
+- `OK` on success.
+- `INVALID_MEMORY_ACCESS` when `name_data`, `name_size`
+  and/or `return_queue_id` point to invalid memory address.
+
+
+#### `proxy_resolve_shared_queue`
+
+* params:
+  - `i32 (const char *) vm_id_data`
+  - `i32 (size_t) vm_id_size`
+  - `i32 (const char *) name_data`
+  - `i32 (size_t) name_size`
+  - `i32 (uint32_t *) return_queue_id`
+* returns:
+  - `i32 (`[`proxy_status_t`]`) status`
+
+Resolves existing shared queue using the provided VM ID (`vm_id_data`,
```

Comment:
can we document what `VM ID` mean here?  this is the first appearance.

Reply:
It's opaque name, it doesn't mean anything, as long as it's used consistently.